src/components/DataTable.tsx

```diff
@@ -233,6 +238,16 @@ export function DataTable<T extends object>(props: Props<T>) {
       });
   };
 
+  useEffect(() => {
+    if (props.includeObjectSelection && data) {
+      const filteredSelectedResources = data.data.data.filter((resource: any) =>
+        selected.includes(resource.id)
```

Comment:
@beganovich I'm not sure if it is possible to pass the type checker when defining the type for the resource. However, I have tried using `as keyof typeof resource` or `as keyof T`, but it does not pass the checker. In this case, I believe it should be written as follows.